packages/veritone-react-common/src/components/SourceManagementForm/index.js

```diff
@@ -0,0 +1,267 @@
+      details: objectOf(any),
+      thumbnailUrl: string,
+    }),
+    initialTemplates: objectOf(
+      shape({
+        id: string,
+        name: string.isRequired,
+        status: string,
+        definition: objectOf(any),
+        data: objectOf(any)
+      })
+    ),
+    onSubmit: func.isRequired,
+    onClose: func.isRequired,
+    open: bool
+  };
+
+  static defaultProps = {
+    initialTemplates: {}
+  };
+
+  state = {
+    selectedSource: null,
+    sourceConfig: {
+      sourceTypeId: '',
+      name: '',
+      thumbnailUrl: '',
+      details: {},
+      thumbnailFile: null
+    },
+    contentTemplates: {},
+    activeTab: 0,
+    openDialog: true
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  UNSAFE_componentWillMount() {
+    const { sourceTypes } = this.props;
+    const newState = {
+      contentTemplates: { ...this.props.initialTemplates }
+    };
+
+    if (has(this.props, 'open')) {
+      newState.openDialog = this.props.open;
+    }
+    if (this.props.source) {
+      // if editing a source, initialize the defaults
+      newState.sourceConfig = {
+        ...pick(this.props.source, ['name', 'thumbnailUrl', 'details']),
+        sourceTypeId: this.props.source.sourceType.id
+      };
+    } else {
+      // If there is no source, then just pick the first available sourceType
+      const fieldValues = {};
+      const sourceTypeIdx = 0;
+      const properties = get(sourceTypes, [
+        sourceTypeIdx,
+        'sourceSchema',
+        'definition',
+        'properties'
+      ]);
+
+      if (properties) {
+        Object.keys(properties).forEach(field => {
+          fieldValues[field] = '';
+        });
+      }
+
+      newState.sourceConfig = {
+        ...this.state.sourceConfig,
+        sourceTypeId: sourceTypes[sourceTypeIdx].id,
+        details: {
+          ...fieldValues
+        }
+      };
+    }
+
+    return this.setState(newState);
+  }
+
+  handleOnClose = () => {
+    return this.setState({ openDialog: false }, () => {
+      this.props.onClose();
+    });
+  };
+
+  handleChangeTab = (e, tabIdx) => {
+    return this.setState({ activeTab: tabIdx });
+  };
+
+  saveConfiguration = config => {
+    return this.setState(prevState => ({
+      sourceConfig: {
+        ...prevState.sourceConfig,
+        ...config
+      }
+    }));
+  };
+
+  manageTemplatesList = (templateSchemaId, remove = false) => {
+    const { templateData, initialTemplates } = this.props;
+
+    if (remove) {
+      if (this.state.contentTemplates[templateSchemaId]) {
+        return this.setState(prevState => {
+          const contentTemplates = { ...prevState.contentTemplates };
+          delete contentTemplates[templateSchemaId];
+
+          return { contentTemplates };
+        });
+      }
+    } else {
+      const data = {};
+      Object.keys(templateData[templateSchemaId].definition.properties).reduce(
+        (fields, schemaDefProp) => {
+          data[schemaDefProp] =
```

Comment:
simplify with get (see my comment in the previous PR)

Reply:
fixed